File: Backend/Microservices/UsersService/UsersService.Domain/Entities/UserWorkoutProgram.cs
using Common.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UsersService.Domain.Entities;

public class UserWorkoutProgram : AuditableBaseEntity
{
  public string Title { get; set; } = default!;
  public string Description { get; set; } = default!;
  public string Content { get; set; } = default!;
}

File: MobileApp/gymly/lib/models/user_workout_program.dart
class UserWorkoutProgram {
  final int id;
  final String title;
  final String description;
  final String content;

  UserWorkoutProgram(
    this.id,
    this.title,
    this.description,
    this.content,
  );

  factory UserWorkoutProgram.fromJson(Map<String, dynamic> json) {
    return UserWorkoutProgram(
      json["id"] as int,
      json["title"] as String,
      json["description"] as String,
      json["content"] as String,
    );
  }
}
